app/models/rubygem.rb

```diff
@@ -286,6 +286,14 @@ def reverse_runtime_dependencies
     reverse_dependencies.where("d.scope ='runtime'")
   end
 
+  def mfa_required?
+    versions.last&.rubygems_mfa_required?
```

Comment:
I'm trying to understand the `versions.last&.rubygems_mfa_required?` check.

Can you clarify, when you push a new version, which is the first version to set rubygems_mfa_required=true, does that version itself require MFA? Or would it only be the next and subsequent versions?

E.g.

- v1.0 does not require MFA
- v1.1 sets rubygems_mfa_required=true.
- v1.2 will require MFA

Would v1.1 also require MFA? Assuming `Rubygem#mfa_required?` is checking against v1.0 which does not have rubygems_mfa_required=true.

Reply:
The intention was to check the latest available version, meaning the metadata always has to be set at each new version. As I am looking at the code now, the method body should probably be `latest_version&.mfa_required`.

Reply:
Thanks @jzajpt. So continuing the example above, when v1.1 is about to be pushed, the latest version is v1.0 (rubygems_mfa_required=nil). Does this mean v1.1 would not require MFA?

Reply:
@eliotsykes Sorry, I deleted my original comment as I realized I had not answered your question. When you're pushing the first version with`rubygems_mfa_required=true`, then the push itself doesn't require MFA yet as the check happens on the latest available version, not the one you're pushing. Only any subsequent action will require it. I will add a test case to cover this scenario.